### PR TITLE
webgpu: support EluGrad kernel

### PR DESCRIPTION
**Pull request by xhcao**
To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

- - -
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/7197)

### CODE REVIEW

File: tfjs-backend-webgpu/src/binary_op_util.ts [hunk 1]
```diff
@@ -59,6 +60,14 @@ const ADD = 'return a + b;';
 const COMPLEX_MULTIPLY_REAL = 'return areal * breal - aimag * bimag;';
 const COMPLEX_MULTIPLY_IMAG = 'return areal * bimag + aimag * breal;';
 const DIV = 'return a / b;';
+const ELU_DER = `
+  let bGTEZero = f32(b > 0.);
```
Comment (gyagp) [inline tfjs-backend-webgpu/src/binary_op_util.ts:64]:
```suggestion
  let bGTEZero = f32(b >= 0.);
```

File: tfjs-backend-webgpu/src/binary_op_util.ts [hunk 2]
```diff
@@ -59,6 +60,14 @@ const ADD = 'return a + b;';
...
+  return (bGTEZero * a) + (1.0 - bGTEZero) * (a * (b + 1.0));
```
Comment (gyagp) [inline tfjs-backend-webgpu/src/binary_op_util.ts:65]: Use select can be simpler

File: tfjs-backend-webgpu/src/binary_op_util.ts [hunk 3]
```diff
@@ -59,6 +60,14 @@ const ADD = 'return a + b;';
...
+`;
+const ELU_DER_VEC4 = `
+  let bGTEZero = vec4<f32>(b >= vec4<f32>(0.));
+  return (bGTEZero * a) + (vec4<f32>(1.0) - bGTEZero) * (a * (b + vec4<f32>(1.0)));
```
Comment (gyagp) [inline tfjs-backend-webgpu/src/binary_op_util.ts:69]: same here, please use select

**Review by gyagp — APPROVED** [review]
LGTM

File: tfjs-backend-webgpu/src/binary_op_util.ts [hunk 4]
```diff
@@ -59,6 +60,9 @@ const ADD = 'return a + b;';
 const COMPLEX_MULTIPLY_REAL = 'return areal * breal - aimag * bimag;';
 const COMPLEX_MULTIPLY_IMAG = 'return areal * bimag + aimag * breal;';
 const DIV = 'return a / b;';
+const ELU_DER = 'return select(a * (b + 1.0), a, b >= 0.);';
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/binary_op_util.ts:63]: I see webgl is using `b >= 1.0` https://github.com/tensorflow/tfjs/blob/master/tfjs-backend-webgl/src/kernels/EluGrad.ts#L24 . However, you are using `b >= 0.0`. Which one is correct?

Comment (xhcao) [reply tfjs-backend-webgpu/src/binary_op_util.ts:63]: On webgl backend, the non PACKED and PACKED versions are different, https://github.com/tensorflow/tfjs/blob/b6a5e25f2517149a13383aaac641f119763968b0/tfjs-backend-webgl/src/kernels/EluGrad.ts#L25
I think the PACKED version is correct, but is not 100% sure.

Comment (gyagp) [reply tfjs-backend-webgpu/src/binary_op_util.ts:63]: According to http://bytedeco.org/javacpp-presets/tensorflow/apidocs/org/bytedeco/tensorflow/EluGrad.html, b >= 0. is correct. 
Please also fix the WebGL unpacked version, and I think we also need some test cases to cover.

Comment (xhcao) [reply tfjs-backend-webgpu/src/binary_op_util.ts:63]: Fixed them and added case in https://github.com/tensorflow/tfjs/pull/7251

Comment (gyagp) [reply tfjs-backend-webgpu/src/binary_op_util.ts:63]: Thanks! Please hold this a bit until #7251 is reviewed.

File: tfjs-backend-webgpu/src/kernels/EluGrad.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC.
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/EluGrad.ts:3]: nit: `2022` -> `2023`

**Review by qjia7 — APPROVED** [review]
LGTM with one nit. Thanks!

**Pull request merged**